sda/service.go

```diff
@@ -359,43 +375,34 @@ func CreateServiceMessage(service string, r interface{}) (*InterServiceMessage,
 
 }
 
-// Client is a simple client structure to be used when wanting to connect to services. It
-// holds the private and public key and allows to connect to a service through
-// the network.
-// The error-handling is done using the ErrorRet structure which can be returned
-// in place of the standard reply. The Client.Send method will catch that and return
-// the appropriate error.
+// Client is a struct used to communicate with a remote Service running on a
+// sda.Conode
 type Client struct {
-	host      *network.SecureTCPHost
 	ServiceID ServiceID
-	sync.Mutex
+	net       *network.Client
 }
 
-// NewClient returns a random client using the service s
+// NewClient returns a client using the service s. It uses TCP communication by
+// default
 func NewClient(s string) *Client {
 	return &Client{
 		ServiceID: ServiceFactory.ServiceID(s),
+		net:       network.NewTCPClient(),
 	}
 }
 
-// Send opens the connection to 'dst' and sends the message 'req'. The
-// reply is returned, or an error if the timeout of 10 seconds is reached.
-func (c *Client) Send(dst *network.ServerIdentity, msg network.Body) (*network.Packet, error) {
-	c.Lock()
-	defer c.Unlock()
-	if c.host == nil {
-		kp := config.NewKeyPair(network.Suite)
-		c.host = network.NewSecureTCPHost(kp.Secret,
-			network.NewServerIdentity(kp.Public, ""))
+// NewLocalClient is for test mainly, it uses the local coomunication network
+// offered by the network package.
+func NewLocalClient(s string) *Client {
```

Comment:
It will for the next PRs for all services that we have where we tests the api, we need to emulate a client using local connections too.